doc/proposals/improved-scorecard-config.md

```diff
@@ -0,0 +1,112 @@
+
+## Goals
+
+Drastically reduce the amount of command line flags in the `scorecard` subcommand and define a configuration file
+format that can allow for the simple configuration of both internal and external plugins.
+
+## Design overview
+
+The configuration for the scorecard will be in a `scorecard` subsection of the config file. This will allow
+the config file to continue working properly when the SDK adds global config file support for all subcommands.
+Under the scorecard subsection there are configuration options that apply to the entire scorecard as well as
+a section that allows for the configuration of both internal and external plugins. Here are the configu options
+and what they do:
+
+- `kubeconfig` string - path to kubeconfig. This option sets the kubeconfig for internal plugins via viper and sets the `KUBECONFIG` env var for external plugins
+- `output` string - sets output format. Valid values: `human-readable` and `json`
+- `plugin-dir` string - path to scorecard plugin directory. This is the directory where the plugins are run from, and all executable files in a `bin` subdirectory of the `plugin-dir` are automtically run by default.
+- `plugins` - this is an array of elements that allow for the configuration of both internal and external scorecard plugins
+
+The elements in the `plugins` configuration have 3 elements: `name`, `disabled`, and either `internal` or `external`. The `name` is
+the name of the plugin and `internal` and `external` are configuration blocks used if the plugin in `internal` or `external`
+respectively. If `internal` and `external` are both specified for the same block, the plugin is automatically marked as failed. The
+`disabled` field is `false` by default, but can be set to `true` to disable tests that would be automatically run, such as the
+`basic` and `olm` tests and external plugins in `plugin-dir/bin`. For internal plugins, the `type` field must be set in
+the`internal` config block and for external plugins the `command` field must be set in the `external` config block for the`disabled`
+config to work properly.
+
+Configuration for `internal` contains all of the original configuration options for the scorecard that pertained to internal plugins. They are:
+
+- `namespace`
+- `init-timeout`
+- `olm-deployed`
+- `csv-path`
+- `namespaced-manifest`
+- `global-manifest`
+- `cr-manifest`
+- `proxy-image`
+- `proxy-pull-policy`
+- `crds-dir`
+
+There is also another config option called `type` which can be set to `basic` or `olm` to configure which internal plugin to run.
+
+Configuration for `external` contains 3 fields:
+
+- `command` string - path to the command being run. Can be relative or absolute. If an executable from `plugin-dir/bin` is specified in the `command` field, it will not be automatically run without configuration as it would be otherwise. The same command can be specified in multiple plugins if a user wishes to run the same plugin multiple times with different configurations.
```

Comment:
```suggestion
- `command` string - path to the command being run. Can be relative or absolute. If an executable from `{plugin-dir}/bin` is specified in the `command` field, it will not be automatically run without configuration as it would be otherwise. The same command can be specified in multiple plugins if a user wishes to run the same plugin multiple times with different configurations.
```